ninja.lua

```diff
@@ -269,7 +285,7 @@ function ninja.generateProjectCfg(cfg)
 		elseif path.iscppfile(node.abspath) then
 			objfilename = obj_dir .. "/" .. node.objname .. intermediateExt(cfg, "cxx")
 			objfiles[#objfiles + 1] = objfilename
-			srcpath = project.getrelative(cfg.workspace, cfg.location .. "/" .. node.relpath)
+			srcpath = project.getrelative(cfg.workspace, "/" .. node.relpath)
```

Comment:
It doesn't compile premake5 without cfg.location because if you refer to sub project which is in another folder, then you need a relative path to from that project workspace in ninja :( Just try to compile premake5 on Linux and you will see.